src/intra_edge.rs

```diff
@@ -214,34 +270,18 @@ impl<const SB128: bool, const N_BRANCH: usize, const N_TIP: usize>
         top_has_right: bool,
         left_has_bottom: bool,
     ) -> (Self, EdgeIndices) {
-        let mut branch = EdgeBranch::new(
-            (if top_has_right {
-                EDGE_TOP_HAS_RIGHT
-            } else {
-                0 as EdgeFlags
-            }) | (if left_has_bottom {
-                EDGE_LEFT_HAS_BOTTOM
-            } else {
-                0 as EdgeFlags
-            }),
-            bl,
-        );
+        let thr = EdgeFlags::TOP_HAS_RIGHT.select(top_has_right);
+        let lhb = EdgeFlags::LEFT_HAS_BOTTOM.select(left_has_bottom);
+        let mut branch = EdgeBranch::new(thr.union(lhb), bl);
         if bl == BL_16X16 {
             let mut n = 0;
             while n < B as u8 {
                 let (tip, next) = indices.tip.pop_front();
                 indices.tip = next;
                 branch.split[n as usize] = tip;
-                let edge_flags = (if n == 3 || (n == 1 && !top_has_right) {
-                    0 as EdgeFlags
-                } else {
-                    EDGE_TOP_HAS_RIGHT
-                }) | (if !(n == 0 || (n == 2 && left_has_bottom)) {
-                    0 as EdgeFlags
-                } else {
-                    EDGE_LEFT_HAS_BOTTOM
-                });
-                self.tip[tip.index as usize] = EdgeTip::new(edge_flags);
+                let thr = EdgeFlags::TOP_HAS_RIGHT.select(!(n == 3 || (n == 1 && !top_has_right)));
+                let lhb = EdgeFlags::LEFT_HAS_BOTTOM.select(n == 0 || n == 2 && left_has_bottom);
```

Comment:
Could you put parens around `n == 2 && left_has_bottom` to make order of operations more clear?

```suggestion
                let lhb = EdgeFlags::LEFT_HAS_BOTTOM.select(n == 0 || n == (2 && left_has_bottom));
```